File: backend/src/Organi.Server/Organi.Server.Domain/Entities/Order.cs
using Organi.Server.Domain.Common;
using Organi.Server.Domain.Enums;

namespace Organi.Server.Domain.Entities;

public sealed class Order : AuditableEntity
{
    public string OrderNumber { get; set; } = default!;
    public decimal SubTotal { get; set; }
    public decimal DiscountAmount { get; set; }
    public decimal ShippingCost { get; set; }
    public decimal TaxAmount { get; set; }
    public decimal TotalAmount { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    public string? Notes { get; set; }
    public string? CancellationReason { get; set; }
    public DateTime? CancelledAt { get; set; }
    public string ShippingFirstName { get; set; } = default!;
    public string ShippingLastName { get; set; } = default!;
    public string ShippingAddress { get; set; } = default!;
    public string ShippingCity { get; set; } = default!;
    public string? ShippingPostalCode { get; set; }
    public string ShippingPhone { get; set; } = default!;
    public string ShippingEmail { get; set; } = default!;
    public Guid UserId { get; set; }
    public Guid? CouponId { get; set; }

    public User User { get; set; } = default!;
    public Coupon? Coupon { get; set; }
    public ICollection<OrderItem> OrderItems { get; set; } = [];
}

File: frontend/admin/src/types/api/order.ts
// Mirrors Organi.Server.Domain.Enums.OrderStatus (serialized as string).
// Processing and Refunded exist in the enum but no endpoint transitions into them.
export type OrderStatus = 'Pending' | 'Confirmed' | 'Processing' | 'Shipped' | 'Delivered' | 'Cancelled' | 'Refunded'

// Mirrors Organi.Server.Application.Features.Orders.DTOs.OrderSummaryResponse
export interface OrderSummaryResponse {
  id: string
  orderNumber: string
  customerName: string
  totalAmount: number
  status: OrderStatus
  itemCount: number
  createdAt: string
}

// Mirrors OrderItemResponse
export interface OrderItemResponse {
  id: string
  productId: string
  productName: string
  productSKU: string
  vendorId: string
  quantity: number
  unitPrice: number
  totalPrice: number
}

// Mirrors OrderResponse (full detail)
export interface OrderResponse {
  id: string
  orderNumber: string
  subTotal: number
  discountAmount: number
  shippingCost: number
  taxAmount: number
  totalAmount: number
  status: OrderStatus
  notes: string | null
  cancellationReason: string | null
  cancelledAt: string | null
  shippingFirstName: string
  shippingLastName: string
  shippingAddress: string
  shippingCity: string
  shippingPostalCode: string | null
  shippingPhone: string
  shippingEmail: string
  userId: string
  couponId: string | null
  items: OrderItemResponse[]
  createdAt: string
}
